File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    const result = number * (number + 1) / 2;
    return result * result;
}

pub fn sumOfSquares(number: usize) usize {
    return (number * (number + 1) * (2 * number + 1)) / 6;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number):
    return (number * (number + 1) // 2) ** 2


def sum_of_squares(number):
    return (number * (number + 1) * (2 * number + 1)) / 6


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
